docling/models/api_vlm_model.py

```diff
@@ -56,9 +53,14 @@ def _vlm_request(page):
                         if hi_res_image.mode != "RGB":
                             hi_res_image = hi_res_image.convert("RGB")
 
+                    if callable(self.vlm_options.prompt):
+                        prompt = self.vlm_options.prompt(page)
```

Comment:
Pass `page.parsed_page` here when using `SegmentedPage` in the callable signature.

Reply:
That was indeed the missing piece, thanks.